### DISCUSSION TITLE
Store Form input: Checkboxes & RadioButtons in LocalStorageState

### DISCUSSION THREAD

**Discussion by SarahWett** [neuefische/questions#537] [Web] [answered]
Hey there,

I created a [multi-step form](https://github.com/SarahWett/capstone-project/pull/8) which contains OneQuestion with **RadioButtons** and one with **Checkboxes** - I can't solve the issue of **saving those inputs inside a localstoragestate**. 

If the buttons Prev/ Next are used, the input is lost (whereas at the textarea it seems to work out).

Obviously I tried to save it with different states and localstoragestate, but none of it works out..
e.g. `  const [state, setState] = useState({
    family: false,
    sports: false,
    walk: false,
    friends: false,
    work: false,
    partner: false,
    hobby: false,
  });

  useEffect(() => {
    const savedState = localStorage.getItem("tagsState");
    if (savedState) {
      setState(JSON.parse(savedState));
    }
  }, []);

  useEffect(() => {
    localStorage.setItem("tagsState", JSON.stringify(state));
  }, [state]);

  const handleInputChange = (e) => {
    const { name, checked } = e.target;
    setState((prevState) => ({ ...prevState, [name]: checked }));
  };`

Searching results all care about text fields only or use local storage with class components.

Thankful for any tipps! :) 

([Clickable on Vercel)](https://capstone-project-7r5jxbbt5-sarahwett.vercel.app/survey)

**Comment by Kheirah** [comment]
I'm afraid I'm missing the question in here :-)

You haven't told us what you've tried thus far. For that reason I can't tell why the things you have tried aren't working.
But: why are trying to store inputs into local storage? Wouldn't a state be enough?

**Comment by marcbendixen** [comment, marked as answer]
Hey @SarahWett, to achieve what you want you need to make a few changes:

Currently the values of the radio buttons are empty (for example: `value={formData.awesome}`) when you fire `onChange={(event) =>
            setFormData({ ...formData, smiley: event.target.value })
          }`
Therefore you need to add a value to each radio button, for example:

```jsx
<label htmlFor="awesome">
        Awesome
        <input
          value="Awesome"
          type="radio"
          id="awesome"
          name="radio"
          onClick={handleToggle}
          onChange={(event) =>
            setFormData({ ...formData, smiley: event.target.value })
          }
        />
      </label>
```

The next step would be to check what your `formData.smiley` is to activate the corresponding radio button when navigating back, example: `checked={formData.smiley === "Awesome"}`

```jsx
<label htmlFor="awesome">
        Awesome
        <input
          value="Awesome"
          type="radio"
          id="awesome"
          name="radio"
          checked={formData.smiley === "Awesome"}
          onClick={handleToggle}
          onChange={(event) =>
            setFormData({ ...formData, smiley: event.target.value })
          }
        />
      </label>
```

**Reply by SarahWett** [reply]
Oh well, this totally makes sense! 
Thank you very much for ending my two-days-horrortrip! :)
